mlflow/sagemaker/__init__.py

```diff
@@ -157,8 +170,35 @@ def deploy(app_name, model_path, execution_role_arn=None, bucket=None, run_id=No
     :param run_id: MLflow run id.
     :param image: name of the Docker image to be used. if not specified, uses a 
                   publicly-available pre-built image.
-    :param region_name: Name of the AWS region to deploy to.
+    :param region_name: Name of the AWS region to which to deploy the application.
+    :param mode: The mode in which to deploy the application. Must be one of the following:
+
+                 mlflow.sagemaker.DEPLOYMENT_MODE_CREATE: Creates an application with the specified 
+                     name and model. This will fail if an application of the same name already 
+                     exists.
+
+                 mlflow.sagemaker.DEPLOYMENT_MODE_REPLACE: Creates an application with the 
+                     specified name and model. This will replace any pre-existing applications of
+                     the same name.
+
+                 mlflow.sagemaker.DEPLOYMENT_MODE_ADD: Adds the specified model to a pre-existing 
+                     application with the specified name, if one exists. If the application does 
+                     not exist,  a new application will be created with the specified name and 
+                     model. NOTE: The specified model will be added to the application's 
```

Comment:
if the endpoint doesn't exist, the model weight will be 1, right? we should document that, or make it 0 (which may not be possible - endpoint creation might fail?).